fl4health/checkpointing/checkpointer.py

```diff
@@ -95,7 +95,12 @@ def maybe_checkpoint(self, model: nn.Module, loss: float, metrics: Dict[str, Sca
                 f"{self.comparison_str} Best score ({self.best_score})",
             )
             self.best_score = comparison_score
-            torch.save(model, self.best_checkpoint_path)
+            try:
+                log(INFO, f"Saving checkpoint as {str(self.best_checkpoint_path)}")
+                torch.save(model, self.best_checkpoint_path)
+            except Exception as e:
+                log(ERROR, f"Encountered the following error while saving the checkpoint: {e}")
+                raise e
```

Comment:
In experimentation, I was encountering an issue where saving the checkpoint would fail, but it wasn't surfacing properly in the logs. So this explicitly places it there.